src/net_processing.cpp

```diff
@@ -1632,6 +1635,38 @@ void PeerManagerImpl::ReattemptInitialBroadcast(CScheduler& scheduler)
     scheduler.scheduleFromNow([&] { ReattemptInitialBroadcast(scheduler); }, delta);
 }
 
+void PeerManagerImpl::ReattemptPrivateBroadcast(CScheduler& scheduler)
+{
+    // Remove stale transactions that are no longer relevant (e.g. already in
+    // the mempool or mined) and count the remaining ones.
+    size_t num_for_rebroadcast{0};
+    const auto stale_txs = m_tx_for_private_broadcast.GetStale();
+    if (!stale_txs.empty()) {
+        LOCK(cs_main);
+        for (const auto& stale_tx : stale_txs) {
+            auto mempool_acceptable = m_chainman.ProcessTransaction(stale_tx, /*test_accept=*/true);
+            if (mempool_acceptable.m_result_type == MempoolAcceptResult::ResultType::VALID) {
+                LogDebug(BCLog::PRIVBROADCAST,
+                         "Reattempting broadcast of stale txid=%s wtxid=%s",
+                         stale_tx->GetHash().ToString(), stale_tx->GetWitnessHash().ToString());
+                ++num_for_rebroadcast;
+            } else {
+                LogPrintLevel(BCLog::PRIVBROADCAST, BCLog::Level::Info,
+                              "Giving up broadcast attempts for txid=%s wtxid=%s: %s",
+                              stale_tx->GetHash().ToString(), stale_tx->GetWitnessHash().ToString(),
+                              mempool_acceptable.m_state.ToString());
+                m_tx_for_private_broadcast.Remove(stale_tx);
+            }
+        }
+    }
+
+    // This could overshoot, but that is ok - we will open some private connections in vain.
+    m_connman.m_private_broadcast.NumToOpenAdd(num_for_rebroadcast);
```

Comment:
nit: this could be moved under the `if` above - seems unnecessary to trigger `notify_all` if there are not stale transactions.